### PR TITLE
feat: navigate - automated tests

### PR DESCRIPTION
**Pull request by renatagandrade**
### Related Issues

- #500
- #860

### Description and Example

Creates automated tests for navigation actions with and without expressions, including:

- pop actions
- push actions with valid and invalid routes
- reset actions with valid and invalid routes

### Checklist

Please, check if these important points are met using `[x]`:

- [X] I read the [PR Guide] and followed the process outlined there for submitting this PR.
- [X] I avoided _breaking changes_ by not changing public APIs that people rely on. 
- [X] I am willing to follow-up on review comments in a timely manner.

### CODE REVIEW

File: backend/automated-tests/src/main/kotlin/br/com/zup/beagle/automatedtests/builders/GlobalSampleTextScreenBuilder.kt [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+
+package br.com.zup.beagle.automatedtests.builders
+
+import NAVIGATE_ACTIONS_ENDPOINT
+import NAVIGATE_RESET_ENDPOINT
+import br.com.zup.beagle.widget.action.Navigate
+import br.com.zup.beagle.widget.action.Route
+import br.com.zup.beagle.widget.context.ContextData
+import br.com.zup.beagle.widget.context.expressionOf
+import br.com.zup.beagle.widget.layout.*
```
Comment (botbeaglezup) [inline backend/automated-tests/src/main/kotlin/br/com/zup/beagle/automatedtests/builders/GlobalSampleTextScreenBuilder.kt:25]: <table data-meta="generated_by_backend">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/ZupIT/beagle/blob/cdc9760c3f5aff402d7f15145648d2b2da6f2ebb/backend/automated-tests/src/main/kotlin/br/com/zup/beagle/automatedtests/builders/GlobalSampleTextScreenBuilder.kt#L25"></span>br.com.zup.beagle.widget.layout.* is a wildcard import. Replace it with fully qualified imports.</td>
    </tr>
  </tbody>
</table>

File: backend/automated-tests/src/main/kotlin/br/com/zup/beagle/automatedtests/builders/GlobalSampleTextScreenBuilder.kt [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+import br.com.zup.beagle.widget.ui.Button
+import br.com.zup.beagle.widget.ui.Text
+
+object GlobalSampleTextScreenBuilder {
+
+    val FALLBACK_SCREEN = NavigateActionsScreenBuilder.build()
+
+    fun build() = Screen(
```
Comment (botbeaglezup) [inline backend/automated-tests/src/main/kotlin/br/com/zup/beagle/automatedtests/builders/GlobalSampleTextScreenBuilder.kt:33]: <table data-meta="generated_by_backend">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/ZupIT/beagle/blob/cdc9760c3f5aff402d7f15145648d2b2da6f2ebb/backend/automated-tests/src/main/kotlin/br/com/zup/beagle/automatedtests/builders/GlobalSampleTextScreenBuilder.kt#L33"></span>The function build is too long (99). The maximum length is 60.</td>
    </tr>
  </tbody>
</table>

File: backend/automated-tests/src/main/kotlin/br/com/zup/beagle/automatedtests/builders/NavigateActionsResetScreenBuilder.kt [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+
+import GLOBAL_TEXT_ENDPOINT
+import GLOBAL_TEXT_EXPRESSION_ENDPOINT
+import NAVIGATE_ACTIONS_ENDPOINT
+import br.com.zup.beagle.widget.action.Navigate
+import br.com.zup.beagle.widget.action.Route
+import br.com.zup.beagle.widget.action.SetContext
+import br.com.zup.beagle.widget.context.ContextData
+import br.com.zup.beagle.widget.context.expressionOf
+import br.com.zup.beagle.widget.layout.*
```
Comment (botbeaglezup) [inline backend/automated-tests/src/main/kotlin/br/com/zup/beagle/automatedtests/builders/NavigateActionsResetScreenBuilder.kt:27]: <table data-meta="generated_by_backend">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/ZupIT/beagle/blob/cdc9760c3f5aff402d7f15145648d2b2da6f2ebb/backend/automated-tests/src/main/kotlin/br/com/zup/beagle/automatedtests/builders/NavigateActionsResetScreenBuilder.kt#L27"></span>br.com.zup.beagle.widget.layout.* is a wildcard import. Replace it with fully qualified imports.</td>
    </tr>
  </tbody>
</table>

File: backend/automated-tests/src/main/kotlin/br/com/zup/beagle/automatedtests/builders/NavigateActionsScreenBuilder.kt [hunk 4]
```diff
@@ -0,0 +1,172 @@
...
+
+package br.com.zup.beagle.automatedtests.builders
+
+import GLOBAL_TEXT_ENDPOINT
+import GLOBAL_TEXT_EXPRESSION_ENDPOINT
+import br.com.zup.beagle.widget.action.Navigate
+import br.com.zup.beagle.widget.action.Route
+import br.com.zup.beagle.widget.action.SetContext
+import br.com.zup.beagle.widget.context.ContextData
+import br.com.zup.beagle.widget.layout.*
```
Comment (botbeaglezup) [inline backend/automated-tests/src/main/kotlin/br/com/zup/beagle/automatedtests/builders/NavigateActionsScreenBuilder.kt:25]: <table data-meta="generated_by_backend">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/ZupIT/beagle/blob/cdc9760c3f5aff402d7f15145648d2b2da6f2ebb/backend/automated-tests/src/main/kotlin/br/com/zup/beagle/automatedtests/builders/NavigateActionsScreenBuilder.kt#L25"></span>br.com.zup.beagle.widget.layout.* is a wildcard import. Replace it with fully qualified imports.</td>
    </tr>
  </tbody>
</table>

File: backend/automated-tests/src/main/kotlin/br/com/zup/beagle/automatedtests/builders/NavigateActionsScreenBuilder.kt [hunk 5]
```diff
@@ -0,0 +1,172 @@
...
+    const val pushStackRemote = "PushStackRemote"
+    const val pushStackRemoteExpression = "PushStackRemoteExpression"
+    const val pushStackRemoteFailure = "PushStackRemoteFailure"
+    const val pushStackRemoteExpressionFailure = "PushStackRemoteExpressionFailure"
+    const val pushViewRemote = "PushViewRemote"
+    const val pushViewRemoteExpression = "PushViewRemoteExpression"
+    const val pushViewRemoteFailure = "PushViewRemoteFailure"
+    const val pushViewRemoteExpressionFailure = "PushViewRemoteExpressionFailure"
+
+    fun build() = Screen(
```
Comment (botbeaglezup) [inline backend/automated-tests/src/main/kotlin/br/com/zup/beagle/automatedtests/builders/NavigateActionsScreenBuilder.kt:40]: <table data-meta="generated_by_backend">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/ZupIT/beagle/blob/cdc9760c3f5aff402d7f15145648d2b2da6f2ebb/backend/automated-tests/src/main/kotlin/br/com/zup/beagle/automatedtests/builders/NavigateActionsScreenBuilder.kt#L40"></span>The function build is too long (132). The maximum length is 60.</td>
    </tr>
  </tbody>
</table>

File: backend/automated-tests/src/main/kotlin/br/com/zup/beagle/automatedtests/controllers/GeneralController.kt [hunk 6]
```diff
@@ -20,16 +20,14 @@ import ACCESSIBILITY_ENDPOINT
...
 import SCREEN_ACTION_CLICK_ENDPOINT
 import SCREEN_SAFE_AREA_FALSE_ENDPOINT
 import SCREEN_SAFE_AREA_TRUE_ENDPOINT
-import br.com.zup.beagle.automatedtests.builders.ClickActionScreenBuilder
-import br.com.zup.beagle.automatedtests.builders.AnalyticsScreenBuilder
-import br.com.zup.beagle.automatedtests.builders.SafeAreaScreenBuilder
-import br.com.zup.beagle.automatedtests.builders.BuilderKotlinDslScreenBuilder
-import br.com.zup.beagle.automatedtests.builders.ExpressionEscapingScreenBuilder
-import br.com.zup.beagle.automatedtests.builders.AccessibilityScreenBuilder
+import br.com.zup.beagle.automatedtests.builders.*
```
Comment (botbeaglezup) [inline backend/automated-tests/src/main/kotlin/br/com/zup/beagle/automatedtests/controllers/GeneralController.kt:30]: <table data-meta="generated_by_backend">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/ZupIT/beagle/blob/cdc9760c3f5aff402d7f15145648d2b2da6f2ebb/backend/automated-tests/src/main/kotlin/br/com/zup/beagle/automatedtests/controllers/GeneralController.kt#L30"></span>br.com.zup.beagle.automatedtests.builders.* is a wildcard import. Replace it with fully qualified imports.</td>
    </tr>
  </tbody>
</table>

**Comment by botbeaglezup** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><a href="https://github.com/ZupIT/beagle/blob/cdc9760c3f5aff402d7f15145648d2b2da6f2ebb/backend/automated-tests/src/main/kotlin/br/com/zup/beagle/automatedtests/controllers/GeneralController.kt#L35">backend/automated-tests/src/main/kotlin/br/com/zup/beagle/automatedtests/controllers/GeneralController.kt#L35</a> - Class ‘GeneralController’ with ‘11’ functions detected. Defined threshold inside classes is set to ‘11’</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_backend">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Comment by botbeaglezup** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">Big PR</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr_danger">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: iOS/Tests/AutomatedTestsUITests/Steps/NavigationActionSteps.swift [hunk 7]
```diff
@@ -0,0 +1,83 @@
...
+//
+//  Created by Lucas Cesar on 05/10/20.
+//  Copyright © 2020 ZUP IT SERVICOS EM TECNOLOGIA E INOVACAO SA. All rights reserved.
+//
+
+import Foundation
+import XCTest
+
+class NavigationActionSteps: CucumberStepsDefinition {
+    var application: XCUIApplication!
```
Comment (botbeaglezup) [inline iOS/Tests/AutomatedTestsUITests/Steps/NavigationActionSteps.swift:13]: <table data-meta="generated_by_ios">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/ZupIT/beagle/blob/cdc9760c3f5aff402d7f15145648d2b2da6f2ebb/iOS/Tests/AutomatedTestsUITests/Steps/NavigationActionSteps.swift#L13"></span>Implicitly unwrapped optionals should be avoided when possible.<br />
<code>implicitly_unwrapped_optional</code> <code>NavigationActionSteps.swift:13</code></td>
    </tr>
  </tbody>
</table>

File: iOS/Tests/AutomatedTestsUITests/Steps/NavigationActionSteps.swift [hunk 8]
```diff
@@ -0,0 +1,83 @@
...
+        }
+        
+        // MARK: - Given
+        Given("^the Beagle application did launch with the navigation screen url$") { _, _ -> Void in
+            XCTAssertTrue(ScreenElements.NAVIGATION_SCREEN_TITLE.element.exists)
+        }
+        
+        // MARK: - When
+        
+        /// Scenarios 1 and 2
```
Comment (botbeaglezup) [inline iOS/Tests/AutomatedTestsUITests/Steps/NavigationActionSteps.swift:31]: <table data-meta="generated_by_ios">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/ZupIT/beagle/blob/cdc9760c3f5aff402d7f15145648d2b2da6f2ebb/iOS/Tests/AutomatedTestsUITests/Steps/NavigationActionSteps.swift#L31"></span>A doc comment should be attached to a declaration.<br />
<code>orphaned_doc_comment</code> <code>NavigationActionSteps.swift:31</code></td>
    </tr>
  </tbody>
</table>

File: iOS/Tests/AutomatedTestsUITests/Steps/NavigationActionSteps.swift [hunk 9]
```diff
@@ -0,0 +1,83 @@
...
+        When("^I press a navigation button \"([^\\\"]*)\"$") { args, _ -> Void in
+            let text = args![0]
```
Comment (botbeaglezup) [inline iOS/Tests/AutomatedTestsUITests/Steps/NavigationActionSteps.swift:33]: <table data-meta="generated_by_ios">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/ZupIT/beagle/blob/cdc9760c3f5aff402d7f15145648d2b2da6f2ebb/iOS/Tests/AutomatedTestsUITests/Steps/NavigationActionSteps.swift#L33"></span>Force unwrapping should be avoided.<br />
<code>force_unwrapping</code> <code>NavigationActionSteps.swift:33</code></td>
    </tr>
  </tbody>
</table>

File: iOS/Tests/AutomatedTestsUITests/Steps/NavigationActionSteps.swift [hunk 10]
```diff
@@ -0,0 +1,83 @@
...
+            self.application.buttons[text].tap()
+        }
+        
+        /// Scenarios 3, 4, 5 and 6
```
Comment (botbeaglezup) [inline iOS/Tests/AutomatedTestsUITests/Steps/NavigationActionSteps.swift:37]: <table data-meta="generated_by_ios">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/ZupIT/beagle/blob/cdc9760c3f5aff402d7f15145648d2b2da6f2ebb/iOS/Tests/AutomatedTestsUITests/Steps/NavigationActionSteps.swift#L37"></span>A doc comment should be attached to a declaration.<br />
<code>orphaned_doc_comment</code> <code>NavigationActionSteps.swift:37</code></td>
    </tr>
  </tbody>
</table>

File: iOS/Tests/AutomatedTestsUITests/Steps/NavigationActionSteps.swift [hunk 11]
```diff
@@ -0,0 +1,83 @@
...
+        When("^I navigate to another screen using the \"([^\\\"]*)\" action and I press a button with the \"([^\\\"]*)\" action$") { args, _ -> Void in
+            let text1 = args![0]
```
Comment (botbeaglezup) [inline iOS/Tests/AutomatedTestsUITests/Steps/NavigationActionSteps.swift:39]: <table data-meta="generated_by_ios">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/ZupIT/beagle/blob/cdc9760c3f5aff402d7f15145648d2b2da6f2ebb/iOS/Tests/AutomatedTestsUITests/Steps/NavigationActionSteps.swift#L39"></span>Force unwrapping should be avoided.<br />
<code>force_unwrapping</code> <code>NavigationActionSteps.swift:39</code></td>
    </tr>
  </tbody>
</table>

File: iOS/Tests/AutomatedTestsUITests/Steps/NavigationActionSteps.swift [hunk 12]
```diff
@@ -0,0 +1,83 @@
...
+            self.application.buttons[text1].tap()
+            XCTAssertTrue(ScreenElements.SAMPLE_NAVIGATION_SCREEN_TITLE.element.exists)
+            let text2 = args![1]
```
Comment (botbeaglezup) [inline iOS/Tests/AutomatedTestsUITests/Steps/NavigationActionSteps.swift:42]: <table data-meta="generated_by_ios">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/ZupIT/beagle/blob/cdc9760c3f5aff402d7f15145648d2b2da6f2ebb/iOS/Tests/AutomatedTestsUITests/Steps/NavigationActionSteps.swift#L42"></span>Force unwrapping should be avoided.<br />
<code>force_unwrapping</code> <code>NavigationActionSteps.swift:42</code></td>
    </tr>
  </tbody>
</table>

File: iOS/Tests/AutomatedTestsUITests/Steps/NavigationActionSteps.swift [hunk 13]
```diff
@@ -0,0 +1,83 @@
...
+            self.application.buttons[text2].tap()
+        }
+        
+        // MARK: - Then
+        
+        /// Scenario 1
```
Comment (botbeaglezup) [inline iOS/Tests/AutomatedTestsUITests/Steps/NavigationActionSteps.swift:48]: <table data-meta="generated_by_ios">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/ZupIT/beagle/blob/cdc9760c3f5aff402d7f15145648d2b2da6f2ebb/iOS/Tests/AutomatedTestsUITests/Steps/NavigationActionSteps.swift#L48"></span>A doc comment should be attached to a declaration.<br />
<code>orphaned_doc_comment</code> <code>NavigationActionSteps.swift:48</code></td>
    </tr>
  </tbody>
</table>

File: iOS/Tests/AutomatedTestsUITests/Steps/NavigationActionSteps.swift [hunk 14]
```diff
@@ -0,0 +1,83 @@
...
+        Then("^the screen should navigate to another screen with the text label \"([^\\\"]*)\"$") { args, _ -> Void in
+            let text = args![0]
```
Comment (botbeaglezup) [inline iOS/Tests/AutomatedTestsUITests/Steps/NavigationActionSteps.swift:50]: <table data-meta="generated_by_ios">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/ZupIT/beagle/blob/cdc9760c3f5aff402d7f15145648d2b2da6f2ebb/iOS/Tests/AutomatedTestsUITests/Steps/NavigationActionSteps.swift#L50"></span>Force unwrapping should be avoided.<br />
<code>force_unwrapping</code> <code>NavigationActionSteps.swift:50</code></td>
    </tr>
  </tbody>
</table>

File: iOS/Tests/AutomatedTestsUITests/Steps/NavigationActionSteps.swift [hunk 15]
```diff
@@ -0,0 +1,83 @@
...
+            XCTAssertTrue(self.application.staticTexts[text].exists)
+        }
+        
+        /// Scenario 2
```
Comment (botbeaglezup) [inline iOS/Tests/AutomatedTestsUITests/Steps/NavigationActionSteps.swift:54]: <table data-meta="generated_by_ios">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/ZupIT/beagle/blob/cdc9760c3f5aff402d7f15145648d2b2da6f2ebb/iOS/Tests/AutomatedTestsUITests/Steps/NavigationActionSteps.swift#L54"></span>A doc comment should be attached to a declaration.<br />
<code>orphaned_doc_comment</code> <code>NavigationActionSteps.swift:54</code></td>
    </tr>
  </tbody>
</table>

File: iOS/Tests/AutomatedTestsUITests/Steps/NavigationActionSteps.swift [hunk 16]
```diff
@@ -0,0 +1,83 @@
...
+        Then("^the screen should not navigate to another screen with the text label \"([^\\\"]*)\"$") { args, _ -> Void in
+            let text = args![0]
```
Comment (botbeaglezup) [inline iOS/Tests/AutomatedTestsUITests/Steps/NavigationActionSteps.swift:56]: <table data-meta="generated_by_ios">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/ZupIT/beagle/blob/cdc9760c3f5aff402d7f15145648d2b2da6f2ebb/iOS/Tests/AutomatedTestsUITests/Steps/NavigationActionSteps.swift#L56"></span>Force unwrapping should be avoided.<br />
<code>force_unwrapping</code> <code>NavigationActionSteps.swift:56</code></td>
    </tr>
  </tbody>
</table>

File: iOS/Tests/AutomatedTestsUITests/Steps/NavigationActionSteps.swift [hunk 17]
```diff
@@ -0,0 +1,83 @@
...
+            XCTAssertFalse(self.application.staticTexts[text].exists)
+        }
+        
+        /// Scenario 3
```
Comment (botbeaglezup) [inline iOS/Tests/AutomatedTestsUITests/Steps/NavigationActionSteps.swift:60]: <table data-meta="generated_by_ios">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/ZupIT/beagle/blob/cdc9760c3f5aff402d7f15145648d2b2da6f2ebb/iOS/Tests/AutomatedTestsUITests/Steps/NavigationActionSteps.swift#L60"></span>A doc comment should be attached to a declaration.<br />
<code>orphaned_doc_comment</code> <code>NavigationActionSteps.swift:60</code></td>
    </tr>
  </tbody>
</table>

File: iOS/Tests/AutomatedTestsUITests/Steps/NavigationActionSteps.swift [hunk 18]
```diff
@@ -0,0 +1,83 @@
...
+        Then("^the app should dismiss the current view$") { _, _ -> Void in
+            XCTAssertFalse(ScreenElements.SAMPLE_NAVIGATION_SCREEN_TITLE.element.exists)
+        }
+
+        /// Scenario 4
```
Comment (botbeaglezup) [inline iOS/Tests/AutomatedTestsUITests/Steps/NavigationActionSteps.swift:65]: <table data-meta="generated_by_ios">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/ZupIT/beagle/blob/cdc9760c3f5aff402d7f15145648d2b2da6f2ebb/iOS/Tests/AutomatedTestsUITests/Steps/NavigationActionSteps.swift#L65"></span>A doc comment should be attached to a declaration.<br />
<code>orphaned_doc_comment</code> <code>NavigationActionSteps.swift:65</code></td>
    </tr>
  </tbody>
</table>

File: iOS/Tests/AutomatedTestsUITests/Steps/NavigationActionSteps.swift [hunk 19]
```diff
@@ -0,0 +1,83 @@
...
+        Then("^the application should navigate back to a specific screen and remove from the stack the other screens loaded from the current screen$") { _, _ -> Void in
+            XCTAssertTrue(ScreenElements.NAVIGATION_SCREEN_TITLE.element.exists)
+            XCTAssertFalse(ScreenElements.SAMPLE_NAVIGATION_SCREEN_TITLE.element.exists)
+        }
+
+        /// Scenario 5
```
Comment (botbeaglezup) [inline iOS/Tests/AutomatedTestsUITests/Steps/NavigationActionSteps.swift:71]: <table data-meta="generated_by_ios">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/ZupIT/beagle/blob/cdc9760c3f5aff402d7f15145648d2b2da6f2ebb/iOS/Tests/AutomatedTestsUITests/Steps/NavigationActionSteps.swift#L71"></span>A doc comment should be attached to a declaration.<br />
<code>orphaned_doc_comment</code> <code>NavigationActionSteps.swift:71</code></td>
    </tr>
  </tbody>
</table>

File: iOS/Tests/AutomatedTestsUITests/Steps/NavigationActionSteps.swift [hunk 20]
```diff
@@ -0,0 +1,83 @@
...
+        Then("^the app should navigate to a specified screen and cleans up the entire stack of the previously loaded views$") { _, _ -> Void in
+            XCTAssertTrue(ScreenElements.RESET_NAVIGATION_SCREEN_TITLE.element.exists)
+            XCTAssertFalse(ScreenElements.NAVIGATION_SCREEN_TITLE.element.exists)
+            XCTAssertFalse(ScreenElements.SAMPLE_NAVIGATION_SCREEN_TITLE.element.exists)
+        }
+        
+        /// Scenario 6
```
Comment (botbeaglezup) [inline iOS/Tests/AutomatedTestsUITests/Steps/NavigationActionSteps.swift:78]: <table data-meta="generated_by_ios">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/ZupIT/beagle/blob/cdc9760c3f5aff402d7f15145648d2b2da6f2ebb/iOS/Tests/AutomatedTestsUITests/Steps/NavigationActionSteps.swift#L78"></span>A doc comment should be attached to a declaration.<br />
<code>orphaned_doc_comment</code> <code>NavigationActionSteps.swift:78</code></td>
    </tr>
  </tbody>
</table>

File: backend/automated-tests/src/main/kotlin/br/com/zup/beagle/automatedtests/builders/GlobalSampleTextScreenBuilder.kt [hunk 21]
```diff
@@ -0,0 +1,143 @@
...
+            listOf(
+                Text(text = "Sample Screen"),
+                Text(text = expressionOf("@{global}")),
+                popButtons(),
+                resetButtons()
+            )
+        )
+    )
+
+    private fun resetButtons(): Container = Container(
```
Comment (botbeaglezup) [inline backend/automated-tests/src/main/kotlin/br/com/zup/beagle/automatedtests/builders/GlobalSampleTextScreenBuilder.kt:51]: <table data-meta="generated_by_backend">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/ZupIT/beagle/blob/7aab42fbe57520592d7617831dc6c542f96f3da9/backend/automated-tests/src/main/kotlin/br/com/zup/beagle/automatedtests/builders/GlobalSampleTextScreenBuilder.kt#L51"></span>The function resetButtons is too long (66). The maximum length is 60.</td>
    </tr>
  </tbody>
</table>

File: android/automated-tests/src/androidTest/kotlin/br/com/zup/beagle/automatedTests/cucumber/elements/ScreenElements.kt [hunk 22]
```diff
@@ -85,3 +85,13 @@ const val SEND_BUTTON = "Enviar"
 const val OK_BUTTON = "OK"
 const val CANCEL_BUTTON = "Cancel"
 
+//Navigate Buttons Screen
```
Comment (uziasferreirazup) [inline android/automated-tests/src/androidTest/kotlin/br/com/zup/beagle/automatedTests/cucumber/elements/ScreenElements.kt:88]: I don't recommend has only one file with all constant.
What do you think about creating a specific file about a specific test?
something like

`NavigationElements.kt`
```

object NavigationElements {

    const val NAVIGATION_SCREEN_TITLE = "Navigation Screen"
    const val PUSH_VIEW_REMOTE_BUTTON = "PushViewRemote"
    const val PUSH_STACK_REMOTE_BUTTON = "PushStackRemote"
    const val PUSH_VIEW_REMOTE_FAILURE_BUTTON = "PushViewRemoteFailure"
    const val PUSH_STACK_REMOTE_FAILURE_BUTTON = "PushStackRemoteFailure"
    const val POP_TO_VIEW_BUTTON = "PopToView"
    const val POP_STACK_BUTTON = "PopStack"
    const val POP_VIEW_BUTTON = "PopView"
    const val RESET_SCREEN_TITLE = "Reset Screen"
}

```

File: android/automated-tests/src/androidTest/kotlin/br/com/zup/beagle/automatedTests/cucumber/steps/Constants.kt [hunk 23]
```diff
@@ -17,6 +17,8 @@
 package br.com.zup.beagle.automatedTests.cucumber.steps
 
 object Constants {
```
Comment (uziasferreirazup) [inline android/automated-tests/src/androidTest/kotlin/br/com/zup/beagle/automatedTests/cucumber/steps/Constants.kt:19]: When you create the constant in kotlin the pattern is all the text in UPPER_CASE, please follow the guideline.
https://kotlinlang.org/docs/reference/coding-conventions.html

File: android/automated-tests/src/androidTest/kotlin/br/com/zup/beagle/automatedTests/cucumber/steps/NavigateScreenSteps.kt [hunk 24]
```diff
@@ -0,0 +1,94 @@
...
+    fun checkGlobalTextScreenIsNotOnView(string2:String) {
+        ScreenRobot()
+            .checkViewDoesNotContainsText(string2)
+    }
+
+    @Then("^the app should dismiss the view that contains (.*)$")
+    fun checkTextIsNotOnAnyView(string1:String){
+        ScreenRobot()
+            .checkViewDoesNotContainsText(string1)
+            .sleep(3)
```
Comment (uziasferreirazup) [inline android/automated-tests/src/androidTest/kotlin/br/com/zup/beagle/automatedTests/cucumber/steps/NavigateScreenSteps.kt:86]: there is some reason to need to add sleep in the code?

File: android/automated-tests/src/main/kotlin/br/com/zup/beagle/automatedTests/activity/AppBeagleActivitiy.kt [hunk 25]
```diff
@@ -68,6 +71,31 @@ class AppBeagleActivitiy : BeagleActivity() {
...
+
+    override fun onDestroy() {
+        super.onDestroy()
+        Log.i("TesteOnPause", "ToOnDestroy $this")
+    }
+
+    // Some function.
+    fun isInForeground(): Boolean {
+        return isInForegroundMode
+    }
```
Comment (uziasferreirazup) [inline android/automated-tests/src/main/kotlin/br/com/zup/beagle/automatedTests/activity/AppBeagleActivitiy.kt:97]: there is some reason to need this code?

**Comment by botbeaglezup** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          2 Warnings
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><a href="https://github.com/ZupIT/beagle/blob/a3ac79d21522544b3f790d105ba909f82b3c7307/android/automated-tests/src/main/AndroidManifest.xml#L8">android/automated-tests/src/main/AndroidManifest.xml#L8</a> - On SDK version 23 and up, your app data will be automatically backed up and restored on app install. Consider adding the attribute <code>android:fullBackupContent</code> to specify an <code>@xml</code> resource which configures which files to backup. More info: https://developer.android.com/training/backup/autosyncapi.html</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><a href="https://github.com/ZupIT/beagle/blob/a3ac79d21522544b3f790d105ba909f82b3c7307/android/automated-tests/src/main/AndroidManifest.xml#L16">android/automated-tests/src/main/AndroidManifest.xml#L16</a> - Attribute <code>usesCleartextTraffic</code> is only used in API level 23 and higher (current min is 19)</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_android">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: android/automated-tests/src/androidTest/assets/features/NavigateScreen.feature [hunk 26]
```diff
@@ -0,0 +1,120 @@
...
+    on the stack and cleans up the navigation that was generated from this screen
+        When I press a navigation button <title>
+        Then the screen should navigate to another screen with the text label <text>
+        And I click on <pop> button
+        Then the app should dismiss the view that contains <text>
+        Then the view that contains the <title> must still exist
+
+        Examples:
+            | title           | text                  | pop       |
+            | PushViewRemote  | PushViewRemoteScreen  | PopToView |
```
Comment (matheusribeirozup) [inline android/automated-tests/src/androidTest/assets/features/NavigateScreen.feature:70]: What do you think of setting up a scenario where the view you want to return does not exist?

**Review by uziasferreirazup — APPROVED** [review]

File: android/automated-tests/src/androidTest/assets/features/NavigateScreen.feature (hunk 26 above)
Comment (carlossteinzup) [reply android/automated-tests/src/androidTest/assets/features/NavigateScreen.feature:70]: Done! I have added this new scenario

**Review by uziasferreirazup — DISMISSED** [review]

**Review by carlossteinzup — APPROVED** [review]

**Review by carlossteinzup — APPROVED** [review]

**Review by matheusribeirozup — APPROVED** [review]

**Review by gabcoelho — APPROVED** [review]

**Pull request merged**